acceptance: remove ExecRoot

tbg opened the pull request:
Trimming the fat and preparing for

https://github.com/cockroachdb/cockroach/pull/17686

tbg committed:
acceptance: phase out ExecRoot

tbg committed:
acceptance: remove (unused) partition tests

We were not actually running these in CI.

tbg committed:
acceptance: remove unused "privileged" flag

tbg committed:
acceptance: remove ExecRoot

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17743)
<!-- Reviewable:end -->

tbg committed:
acceptance: let ExecCLI fill in sane values

Only the specific implementation of `cluster.Cluster` can know these;
the tests should be agnostic to the endpoint.

This should carry us for a while, but at some point we might have to
provide a more direct mapping.

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 4 files at r1, 5 of 5 files at r2, 2 of 2 files at r3, 3 of 3 files at r4, 3 of 3 files at r5.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17743#-:-KrmleLLxdTr8vOVKF-6:bnfp4nl)*
<!-- Sent from Reviewable.io -->
